plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -487,20 +501,29 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
                                                                  read_only ? transaction_metadata::trx_type::read_only : transaction_metadata::trx_type::input,
                                                                  chain.configured_subjective_signature_length_limit() );
 
+         if( !read_only ) {
+            next = [this, trx, next{std::move(next)}]( const std::variant<fc::exception_ptr, transaction_trace_ptr>& response ) {
```

Comment:
Clear `unapplied_transaction_queue` on shutdown so that `this` does not outlive this lambda.